src/main/java/seedu/tr4cker/commons/core/index/Index.java

```diff
@@ -15,7 +15,7 @@
      * Index can only be created by calling {@link Index#fromZeroBased(int)} or
      * {@link Index#fromOneBased(int)}.
      */
-    private Index(int zeroBasedIndex) {
+    public Index(int zeroBasedIndex) {
```

Comment:
should this keep the private access, and methods that use this call fromZeroBased(int) or fromOneBased(int) ?